modules/caddytls/leafstorageloader.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caddytls
+
+import (
+	"crypto/x509"
+	"fmt"
+
+	"github.com/caddyserver/certmagic"
+
+	"github.com/caddyserver/caddy/v2"
+)
+
+func init() {
+	caddy.RegisterModule(LeafStorageLoader{})
+}
+
+// LeafStorageLoader loads leaf certificates from the
+// globally configured storage module.
+type LeafStorageLoader struct {
+	// A list of pairs of certificate file names along with their
+	// encoding format so that they can be loaded from storage.
+	Certs []LeafCertFile `json:"certs,omitempty"`
+
+	// Reference to the globally configured storage module.
+	storage certmagic.Storage
```

Comment:
Add a field for custom storage for this module with default to `ctx.Storage()`. Here's an example:

https://github.com/caddyserver/caddy/blob/e1aa862e6a951d4dc9c0e89b6d42f7fea1c5aba7/modules/caddytls/capools.go#L337-L351